fastcrypto-tbls/src/nodes.rs

```diff
@@ -74,4 +74,33 @@ impl<G: GroupElement> Nodes<G> {
     pub fn iter(&self) -> impl Iterator<Item = &Node<G>> {
         self.nodes.iter()
     }
+
+    /// Reduce weights up to an allowed delta in the original total weight.
+    /// Finds the largest d such that:
+    /// - The new threshold is ceil(t / d)
+    /// - The new weights are all divided by d (floor division)
+    /// - The precision loss, counted as the sum of the remainders of the division by d, is at most
+    ///   the allowed delta
+    /// In practice, allowed delta will be the extra liveness we would assume above 2f+1.
+    pub fn reduce(&self, t: u16, allowed_delta: u16) -> (Self, u16) {
+        let mut max_d = 1;
+        for d in 2..=40 {
+            let sum = self.nodes.iter().map(|n| n.weight % d).sum::<u16>();
+            if sum <= allowed_delta {
+                max_d = d;
+            }
+        }
+        let nodes = self
```

Comment:
I don't know if you win a lot by doing so, but you could exit early here if max_d == 1.